### PR TITLE
ios: notes to self

### PR DESCRIPTION
**Pull request by avently**

### CODE REVIEW

File: apps/ios/Shared/Model/SimpleXAPI.swift [hunk 1]
```diff
@@ -365,6 +365,32 @@ func apiSendMessage(type: ChatType, id: Int64, file: CryptoFile?, quotedItemId:
     }
 }
 
+func apiCreateChatItem(noteFolderId: Int64, file: CryptoFile?, msg: MsgContent) async -> ChatItem? {
+    let chatModel = ChatModel.shared
+    let cmd: ChatCommand = .apiCreateChatItem(noteFolderId: noteFolderId, file: file, msg: msg)
+    var cItem: ChatItem? = nil
+    let endTask = beginBGTask({
+        if let cItem = cItem {
+            DispatchQueue.main.async {
```
Comment (epoberezkin) [inline apps/ios/Shared/Model/SimpleXAPI.swift:374]: MainActor

File: apps/ios/Shared/Model/SimpleXAPI.swift [hunk 2]
```diff
@@ -365,6 +365,32 @@ func apiSendMessage(type: ChatType, id: Int64, file: CryptoFile?, quotedItemId:
     }
 }
 
+func apiCreateChatItem(noteFolderId: Int64, file: CryptoFile?, msg: MsgContent) async -> ChatItem? {
+    let chatModel = ChatModel.shared
+    let cmd: ChatCommand = .apiCreateChatItem(noteFolderId: noteFolderId, file: file, msg: msg)
+    var cItem: ChatItem? = nil
+    let endTask = beginBGTask({
```
Comment (epoberezkin) [inline apps/ios/Shared/Model/SimpleXAPI.swift:372]: this is unnecessary here, it's not sending anything, can be done as other apis without background tasks

File: apps/ios/SimpleX NSE/NotificationService.swift [hunk 3]
```diff
@@ -783,6 +783,9 @@ func autoReceiveFile(_ file: CIFile) -> ChatItem? {
     case .xftp:
         apiSetFileToReceive(fileId: file.fileId, encrypted: encrypted)
         return nil
+    case .local:
+        apiSetFileToReceive(fileId: file.fileId, encrypted: encrypted)
```
Comment (epoberezkin) [inline apps/ios/SimpleX NSE/NotificationService.swift:787]: Don't think it's needed here, we cannot receive local file in NSE

File: apps/ios/SimpleXChat/ChatTypes.swift [hunk 4]
```diff
@@ -2227,7 +2282,7 @@ public struct ChatItem: Identifiable, Decodable {
     }
 
     public var allowAddReaction: Bool {
-        meta.itemDeleted == nil && !isLiveDummy && reactions.filter({ $0.userReacted }).count < 3
+        return meta.itemDeleted == nil && !isLiveDummy && reactions.filter({ $0.userReacted }).count < 3
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/ChatTypes.swift:2285]:
```suggestion
        meta.itemDeleted == nil && !isLiveDummy && reactions.filter({ $0.userReacted }).count < 3
```

File: apps/ios/SimpleXChat/FileUtils.swift [hunk 5]
```diff
@@ -22,6 +22,8 @@ public let MAX_VIDEO_SIZE_AUTO_RCV: Int64 = 1_047_552 // 1023KB
 
 public let MAX_FILE_SIZE_XFTP: Int64 = 1_073_741_824 // 1GB
 
+public let MAX_FILE_SIZE_LOCAL: Int64 = 1_073_741_824 // 1GB
```
Comment (epoberezkin) [inline apps/ios/SimpleXChat/FileUtils.swift:25]: suggest removing limit :)

File: apps/ios/Shared/Views/Chat/ChatItem/CIFileView.swift [hunk 6]
```diff
@@ -97,7 +97,7 @@ struct CIFileView: View {
                 }
```
Comment (spaced4ndy) [inline apps/ios/Shared/Views/Chat/ChatItem/CIFileView.swift:97]: doesn't open save dialogue on tap (not via menu but on chat item tap)

File: apps/ios/Shared/Views/Chat/ChatItem/CIFileView.swift [hunk 7]
```diff
@@ -125,10 +125,11 @@ struct CIFileView: View {
                 switch file.fileProtocol {
                 case .xftp: progressView()
                 case .smp: fileIcon("doc.fill")
+                case .local: fileIcon("doc.fill")
                 }
             case let .sndTransfer(sndProgress, sndTotal):
                 switch file.fileProtocol {
-                case .xftp: progressCircle(sndProgress, sndTotal)
+                case .xftp, .local: progressCircle(sndProgress, sndTotal)
```
Comment (spaced4ndy) [inline apps/ios/Shared/Views/Chat/ChatItem/CIFileView.swift:132]: same as on android - can't be in progress (transferred), can't be waiting (accepted)

**Pull request merged**